File: src/scripts/prebuild/provider.test.ts
import { describe, it, expect } from 'vitest';
import * as path from 'path';
import { removeDiacritics, sanitize, groupAndSaveCSV, executeYamlStructure } from './provider';
describe('removeDiacritics', () => {
    it('removes diacritics from strings', () => {
        expect(removeDiacritics("Ångström")).toBe("Angstrom");
        expect(removeDiacritics("Øresund")).toBe("Oresund");
        expect(removeDiacritics("Æble")).toBe("AEble");
    });
});
describe('sanitize', () => {
    it('sanitizes strings to be filesystem and URL safe', () => {
        expect(sanitize("New York")).toBe("new-york");
        expect(sanitize("São Paulo")).toBe("sao-paulo");
        expect(sanitize("Café")).toBe("cafe");
    });
});
describe('groupAndSaveCSV', () => {
    it('groups data and returns correct structure', () => {
        const data = [
            { Region: "North", Locality: "City A", Niche: "N1", Value: "1" },
            { Region: "North", Locality: "City A", Niche: "N2", Value: "2" },
        ];
        const outputDir = path.join(__dirname, 'output');
        const result = groupAndSaveCSV(data, outputDir, null);
        expect(result).toHaveProperty("North");
        expect(result["North"]).toHaveProperty("City A");
        const details = result["North"]["City A"];
        expect(details.createDirectory).toBe(true);
        expect(details.file).toMatch(/n2\/content\.csv$/i);
    });
});
// describe('executeYamlStructure', () => {
//     it('creates directory and writes CSV files correctly', () => {
//         const tempDir = path.join(__dirname, 'temp');
//         if (fs.existsSync(tempDir)) fs.rmSync(tempDir, { recursive: true });
//         fs.mkdirSync(tempDir, { recursive: true });
//         const data = {
//             North: {
//                 CityA: {
//                     createDirectory: true,
//                     file: path.join(tempDir, 'North/CityA/Niche/content.csv'),
//                     data: [{ Column1: "Value1" }]
//                 }
//             }
//         };
//         executeYamlStructure(data);
//         const filePath = path.join(tempDir, 'North/CityA/Niche/content.csv');
//         expect(fs.existsSync(filePath)).toBe(true);
//         const csvContent = fs.readFileSync(filePath, 'utf-8');
//         const parsed = Papa.parse(csvContent, { header: true }).data;
//         expect(parsed[0]['Column1']).toBe("Value1");
//     });
// });

File: src/scripts/prebuild/provider_test.py
import pandas as pd
from provider import sanitize, remove_diacritics, group_and_save_csv, execute_yaml_structure
def test_remove_diacritics():
    assert remove_diacritics("Ångström") == "Angstrom"
    assert remove_diacritics("Øresund") == "Oresund"
    assert remove_diacritics("Æble") == "AEble"
def test_sanitize():
    assert sanitize("New York") == "new-york"
    assert sanitize("São Paulo") == "sao-paulo"
    assert sanitize("Café") == "cafe"
def test_group_and_save_csv(tmp_path):
    df = pd.DataFrame({
        "Region": ["North", "North"],
        "Locality": ["City A", "City A"],
        "Niche": ["N1", "N2"],
        "Value": [1, 2]
    })
    output_dir = tmp_path / "output"
    result = group_and_save_csv(df, output_dir, None)
    print(result)
    assert "North" in result
    assert "City A" in result["North"]
    details = result["North"]["City A"]
    assert details["create_directory"]
    print(details["file"])
    assert details["file"].endswith("n2/content.csv")
